ext/http/lib.rs

```diff
@@ -425,6 +428,39 @@ impl Default for HttpResponseWriter {
   }
 }
 
+struct BodyCompressedHolder {
```

Comment:
It would be a little cleaner if `BodyCompressedHolder` itself just implemented AsyncWrite.

Reply:
If you don't want to do that and make this essentially a POD struct, why not define:

```rust
enum HttpResponseWriter {
  // ...
  Body { writer: Pin<Box<dyn tokio::io::AsyncWrite>>, is_shut: Arc<AtomicBool> }
  // ...
}
```